SqlInstall.ps1

```diff
@@ -0,0 +1,433 @@
+
+    (Get-Content -Path$configini).Replace('SQLTEMPDBDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"', 'SQLTEMPDBDIR="' + $TempVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Data"') | Out-File $configini
+
+    (Get-Content -Path $configini).Replace('SQLUSERDBLOGDIR="E:\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"', 'SQLUSERDBLOGDIR="' + $LogVolume + ':\Program Files\Microsoft SQL Server\MSSQL12.AXIANSDB01\MSSQL\Log"') | Out-File $configini
+
+    (Get-Content -Path $configini).Replace('SQLSYSADMINACCOUNTS="WIN-NAJQHOBU8QD\Administrator"', 'SQLSYSADMINACCOUNTS="' + $SqlServerAccount)| Out-File $configini
+
+    If ($SaveFile -eq "Yes") {
+        $SaveFileLocation = Read-Host "Please enter your preferred directory for saving a copy of the configuration file: "
+        Copy-Item $configini -Destination $SaveFileLocation
+    }
+
+    IF ($Authentication -eq "Mixed Mode") {
+        $SAPassW = [PsCredential](Get-Credential -UserName "SA"  -Message "Please Enter the SA Password.")
+
+        & $SetupFile /ConfigurationFile=$configini /Q /IACCEPTSQLSERVERLICENSETERMS /SAPWD= $SAPassW.GetNetworkCredential().Password
+
+    }
+    else {
+        & $SetupFile /ConfigurationFile=$configini /Q /IACCEPTSQLSERVERLICENSETERMS
+    }
+
+    
+
+    # Grant service account the right to perform volume maintenance
+    # code found at https://social.technet.microsoft.com/Forums/windows/en-US/5f293595-772e-4d0c-88af-f54e55814223/adding-domain-account-to-the-local-policy-user-rights-assignment-perform-volume-maintenance?forum=winserverpowershell
+
+    if ($PerformVolumeMaintenance) {
+        ## <--- Configure here
+        $accountToAdd = 'NT Service\MSSQL$AXIANSDB01'
+        ## ---> End of Config
+        $sidstr = $null
+
+
+        try {
+            $ntprincipal = new-object System.Security.Principal.NTAccount "$accountToAdd"
+            $sid = $ntprincipal.Translate([System.Security.Principal.SecurityIdentifier])
+            $sidstr = $sid.Value.ToString()
+        }
+        catch {
+            $sidstr = $null
+        }
+        Write-Message -Level Verbose -Message "Account: $($accountToAdd)" -ForegroundColor DarkCyan
+        if ( [string]::IsNullOrEmpty($sidstr) ) {
+            Write-Message -Level Verbose -Message "Account not found!" -ForegroundColor Red
+            #exit -1
+        }
+
+        Write-Message -Level Verbose -Message "Account SID: $($sidstr)" -ForegroundColor DarkCyan
+        $tmp = ""
+        $tmp = [System.IO.Path]::GetTempFileName()
+        Write-Message -Level Verbose -Message "Export current Local Security Policy" -ForegroundColor DarkCyan
+        secedit.exe /export /cfg "$($tmp)" 
+        $c = ""
```

Comment:
Please use the other dbatools commands such as `Set-DbaPrivilege`

Reply:
Now that I think about it, this is likely not in scope. If the SQL Server binaries don't support it, we shouldn't either. If they do, we should rely on the binaries.

Reply:
I think it is in scope. Starting with SQL 2016 you can set a check on perform volume maintenance in the script. So that part has to be in there for 2016 and up. I think it would be a nice service to the users to enable the same functionality for earlier versions. At this point i might need to use other code (Set-DbaPrivilige as you suggest).

Reply:
great point, that'd provide backwards compat for things that are offered in newer binaries. sold